File: Arrays/reverse_a_array_recursively.cpp
#include <iostream>
#include <cstdlib> //for rand()
#include <ctime>   //for time()
#include <cstddef> // For null,
using namespace std;

void reverseArray(int array[], int start, int end)
{
    int temp;
    if (start >= end)
    {
        return;
    }
    temp = array[start];
    array[start] = array[end];
    array[end] = temp;
    reverseArray(array, start + 1, end - 1);
}
int main()
{

    srand(time(NULL));

    int n, *array;
    cout << "Enter length of array.   ";
    cin >> n;

    array = (int *)new int(n);

    for (int i = 0; i < n; i++)
    {
        array[i] = rand() % 200;
    }

    cout << "Before Reversing:  " << endl;
    for (int i = 0; i < n; i++)
    {
        cout << array[i] << "  ";
    }
    cout << "\n";

    reverseArray(array, 0, n - 1);

    cout << "\nAfter Reversing:  " << endl;
    for (int i = 0; i < n; i++)
    {
        cout << array[i] << "  ";
    }
    cout << "\n";
    return 0;
}

File: Arrays/reverse_a_array_recursively.py
import numpy as np


def reverseArray(arr, start, end):
    while start >= end:
        return

    arr[start], arr[end] = arr[end], arr[start]
    start += 1
    end -= 1
    reverseArray(arr, start, end)


n = int(input("Enter the length of array...  "))
array = np.random.randint(0, 250, n, dtype=int)
l = len(array)
print("\nBefore reversing,... \n", array)
reverseArray(array, 0, l - 1)
print("\nAfter reversing,... \n", array)